src/gpodder/gtkui/desktop/preferences.py

```diff
@@ -321,21 +321,37 @@ def new(self):
 
         gpodder.user_extensions.on_ui_object_available('preferences-gtk', self)
 
+        self.inject_extensions_preferences(init=True)
+
+        self.prefs_stack.foreach(self._wrap_checkbox_labels)
+
+    def _wrap_checkbox_labels(self, w, *args):
+        if w.get_name().startswith("no_label_wrap"):
+            return
+        elif isinstance(w, Gtk.CheckButton):
+            label = w.get_child()
+            label.set_line_wrap(True)
+        elif isinstance(w, Gtk.Container):
+            w.foreach(self._wrap_checkbox_labels)
+
+    def inject_extensions_preferences(self, init=False):
+        if not init:
+            # remove preferences buttons for all extensions
+            assert(self.prefs_stack.get_visible_child_name() == 'extensions')
+            found_extensions = False
+            for child in self.prefs_stack.get_children():
+                if found_extensions:
+                    self.prefs_stack.remove(child)
+                elif child == self.prefs_stack.get_visible_child():
```

Comment:
This relies on the 'Extensions' page being the last item in the Stack before the extension pages. Maybe prepend `"extension."` to the `name` of the extension stack children, and remove only those which have this prefix?

Reply:
Do you know how to do this? This is a much better solution but I couldn't find a way to iterate over names, and get_child_by_name() would need the full extension name. I could split the removal code to another function called before the enabled/disabled extension has been added/removed to/from user_preferences, and then iterate over that to remove each extension.

Reply:
Ok, it did not quite work as I thought, but setting the widget name to something recognizable works. See this patch:

```diff
diff --git a/src/gpodder/gtkui/desktop/preferences.py b/src/gpodder/gtkui/desktop/preferences.py
index 68550c13..e84ed898 100644
--- a/src/gpodder/gtkui/desktop/preferences.py
+++ b/src/gpodder/gtkui/desktop/preferences.py
@@ -337,21 +337,19 @@ class gPodderPreferences(BuilderWidget):
     def inject_extensions_preferences(self, init=False):
         if not init:
             # remove preferences buttons for all extensions
-            assert(self.prefs_stack.get_visible_child_name() == 'extensions')
-            found_extensions = False
             for child in self.prefs_stack.get_children():
-                if found_extensions:
+                if child.get_name().startswith("extension."):
                     self.prefs_stack.remove(child)
-                elif child == self.prefs_stack.get_visible_child():
-                    found_extensions = True
 
         # add preferences buttons for all extensions
         result = gpodder.user_extensions.on_preferences()
         if result:
             for label, callback in result:
-                self.prefs_stack.add_titled(callback(), label, label)
-
-            self.prefs_stack.foreach(self._wrap_checkbox_labels)
+                page = callback()
+                name = "extension." + label
+                page.set_name(name)
+                page.foreach(self._wrap_checkbox_labels)
+                self.prefs_stack.add_titled(page, name, label)
 
     def _extensions_select_function(self, selection, model, path, path_currently_selected):
         return model.get_value(model.get_iter(path), self.C_SHOW_TOGGLE)
```

Reply:
Thank you, I have pushed your patch.